Revert "Remove __torch__ from custom class qualname"

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32555 [JIT] Serialize attributes and types in ClassType serialization
* #32553 [JIT] Support for registering C++ lambdas as methods on custom C++ class
* **#32514 Revert "Remove __torch__ from custom class qualname"**
* #32508 [JIT] Add torch.classes.load_library
* #32477 [JIT] Support returning tuple from custom bound C++ method
* #32471 [JIT] Fix custom class method binding for const methods
* #32470 [JIT] Test __getstate__ and __setstate__ for custom bound C++ classes
* #32260 [JIT] Passing custom class as arg

This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

Commit:
Revert "Remove __torch__ from custom class qualname"

This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

[ghstack-poisoned]

Commit:
Update on "Revert "Remove __torch__ from custom class qualname""


This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit b7f34dde:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=b7f34dde5d648014c003a80814665ac8eda21cc6).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4320537) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=89600380) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=89600380))
<details>
<summary>
<code>Automatic merge failed; fix conflicts and then commit the result.</code>
</summary>

```
 
HEAD is now at b7f34dde5d Update on "Revert "Remove __torch__ from custom class qualname"" 
+ git reset --hard b7f34dde5d648014c003a80814665ac8eda21cc6 
HEAD is now at b7f34dde5d Update on "Revert "Remove __torch__ from custom class qualname"" 
+ git merge --allow-unrelated-histories --no-edit --no-ff 193ac3144114de6ff37ba4e8b279ec4d479eac96 
Auto-merging torch/csrc/jit/script/init.cpp 
Auto-merging torch/csrc/jit/pybind_utils.h 
CONFLICT (content): Merge conflict in torch/csrc/jit/pybind_utils.h 
Auto-merging test/test_jit.py 
Removing test/cpp/jit/test_constant_propagation.cpp 
Automatic merge failed; fix conflicts and then commit the result. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "Revert "Remove __torch__ from custom class qualname""


This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

[ghstack-poisoned]

Commit:
Update on "Revert "Remove __torch__ from custom class qualname""


This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

[ghstack-poisoned]

Commit:
Update on "Revert "Remove __torch__ from custom class qualname""


This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

[ghstack-poisoned]

Commit:
Update on "Revert "Remove __torch__ from custom class qualname""


This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

[ghstack-poisoned]

Commit:
Update on "Revert "Remove __torch__ from custom class qualname""


This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

[ghstack-poisoned]

Commit:
Update on "Revert "Remove __torch__ from custom class qualname""


This reverts commit c7fdf5b251c6fecd5d78b4f33d30bd77ca3f841c.

Turns out we actually need this for correct serialization

Differential Revision: [D19525532](https://our.internmc.facebook.com/intern/diff/D19525532)

[ghstack-poisoned]

zdevito commented on test/test_jit.py:
```diff
@@ -4922,6 +4922,18 @@ def f():
+    @unittest.skipIf(IS_WINDOWS, "TODO: Fix this test case")
+    def test_torchbind_save_load(self):
+        def foo():
+            ss = torch.classes._TorchScriptTesting_StackString(["mom"])
```
hi .. mom makes more sense then mom .. hi

jamesr66a replied:
"Mom, hi!"
